GVFS/GVFS.Platform.Windows/DiskLayoutUpgrades/DiskLayout16to17Upgrade_FolderPlaceholderValues.cs

```diff
@@ -0,0 +1,75 @@
+﻿using GVFS.Common;
+using GVFS.Common.FileSystem;
+using GVFS.Common.Tracing;
+using GVFS.DiskLayoutUpgrades;
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace GVFS.Platform.Windows.DiskLayoutUpgrades
+{
+    /// <summary>
+    /// Updates the values for folder placeholders from AllZeroSha to PlaceholderListDatabase.PartialFolderValue
+    /// </summary>
+    public class DiskLayout16to17Upgrade_FolderPlaceholderValues : DiskLayoutUpgrade.MajorUpgrade

```

Comment:
@jrbriggs it would be great to do some manual testing of this upgrade on different starting data shapes

Reply:
I've smoke tested several upgrade scenarios and we'll do more testing once we have real-signed bits.